### DISCUSSION TITLE
`import { Application } from "pixi.js"`  vs.  `import { Application } from "@pixi/app"`

### DISCUSSION THREAD

**Discussion by joergplewe** [pixijs/pixijs#8035] [Q&A] [answered]
Hi all!

I am very confused about the different ways to import Pixi:
* import the lib as a whole as "pixi.js"
* import per package as "@pixi/xxx"

When to use which? What is the difference? At least both options lead to very different behavior e.g. of `Application`.
Can you give hints to the documentation of the two? How to move from one to the other?

Here are two samples:

```
import { Application, Text } from "pixi.js";

const app = new Application({backgroundColor: 0xff8000});
const { renderer, stage } = app;

renderer.on("prerender", () => console.log("prerender"));

document.body.appendChild(app.view);

var txt = new Text("hello", {fontSize: 300});
stage.addChild(txt);
txt.interactive = true;
txt.buttonMode = true;
txt.on("pointerdown", (evt) => console.log("click"));
```
This one works as expected, the renderloop starts automatically and I can click the `Text`.

```
import { Renderer } from '@pixi/core';
import { Application } from '@pixi/app';
import { Text } from '@pixi/text';

import { InteractionManager } from '@pixi/interaction';
Renderer.registerPlugin('interaction', InteractionManager);

const app = new Application({backgroundColor: 0xff8000});
const { renderer, stage } = app;

renderer.on("prerender", () => console.log("prerender"));

document.body.appendChild(app.view);

var txt = new Text("hello", {fontSize: 200});
stage.addChild(txt);
txt.interactive = true;
txt.buttonMode = true;
txt.on("pointerdown", (evt) => console.log("click"));

// Render stage so that it becomes the root target for UI events
renderer.render(stage);
```
In this one, the renderloop does not start, I need to register the `InteractionManager`, the `Text` rendering crashes.
What is missing? And how can I know?

Kindly hope for help,

. J

**Comment by bigtimebuddy** [comment, marked as answer]
Great question @joergplewe, let's see if I can help clarify.

We publish two "bundles" (`pixi.js` and `pixi.js-legacy`) these are just simple entry-points that combine a collection of the `@pixi/*` packages.  We created a tool that allows you to create your own custom bundle: https://pixijs.io/customize

If you are going to use the `@pixi/*` packages, you need to make sure that Application/Renderer/Loader plugins are registered correctly somewhere in your code. Please check out the customize tool above or you can look at the [source code](https://github.com/pixijs/pixijs/blob/dev/bundles/pixi.js/src/index.ts) for `pixi.js`, which is pretty straightforward.

In general, if you aren't super concerned about the filesize footprint of Pixi, just import from `pixi.js` and `pixi.js-legacy` and call it a day. It's going to make things much simpler for your code.

**Reply by joergplewe** [reply]
This is really helpful getting my foot in the door.... thank you!
